cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/api/artifact/ArtifactVersionTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.artifact;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests the parsing of artifact version.
+ */
+public class ArtifactVersionTest {
+
+  @Test
+  public void testValidVersions() {
+    // since a version's suffix can not have numbers in it, simply the number 1 is the version
+    String versionString = "cdap-etl-batch-3.5-SNAPSHOT1";
+    ArtifactVersion artifactVersion = new ArtifactVersion(versionString, true);
+    assertVersionEquals("1", 1, null, null, null, artifactVersion);
+
+    versionString = "s3_analytics-1.0-SNAPSHOT";
+    artifactVersion = new ArtifactVersion(versionString, true);
+    assertVersionEquals("1.0-SNAPSHOT", 1, 0, null, "SNAPSHOT", artifactVersion);
+
+    // because of the restriction that the suffix can not have numbers in it, '3.5' is not to be part of the version
+    versionString = "cdap-etl-3.5-batch-1.0-SNAPSHOT";
+    artifactVersion = new ArtifactVersion(versionString, true);
+    assertVersionEquals("1.0-SNAPSHOT", 1, 0, null, "SNAPSHOT", artifactVersion);
+
+    versionString = "cdap-etl-batch-3.2.0-suffix";
+    artifactVersion = new ArtifactVersion(versionString, true);
+    assertVersionEquals("3.2.0-suffix", 3, 2, 0, "suffix", artifactVersion);
+
+    versionString = "1.0-SNAPSHOT";
+    artifactVersion = new ArtifactVersion(versionString, false);
+    assertVersionEquals("1.0-SNAPSHOT", 1, 0, null, "SNAPSHOT", artifactVersion);
+  }
+
+  @Test
+  public void testInvalidVersions() {
+    // invalid simply because there are no numbers in the string
```

Comment:
Line 65 is an example of something that is an invalid version, but would be valid if numbers were allowed in the suffix.